module/VuFind/src/VuFind/ILS/Driver/XCNCIP2.php

```diff
@@ -1146,26 +1147,34 @@ public function getMyStorageRetrievalRequests($patron)
             $requestId = $current->xpath('ns1:RequestId/ns1:RequestIdentifierValue');
             $requestType = $current->xpath('ns1:RequestType');
             $requestType = (string)$requestType[0];
-            $tmpStatus = $current->xpath('ns1:RequestStatusType');
-            list($status, $created) = explode(" ", (string)$tmpStatus[0], 2);
-            if ($status === "Accepted") {
-                $cancelled = false;
+            $created = $current->xpath('ns1:DatePlaced');
+            $created = !empty($created)
+                ? date("l, d-M-y h:i a", strtotime((string)$created[0])) : null;
+            $requestStatusType = $current->xpath('ns1:RequestStatusType');
+            $status = !empty($requestStatusType) ? (string)$requestStatusType[0]
+                : null;
+            if (!in_array($status, ['Available For Pickup', 'In Process'])) {
```

Comment:
I notice that the original code used 'Accepted' for this condition, but you do not. Is this something that varies by instance? Should the list be configurable somewhere?

Reply:
It's a problem in NCIP standard itself - The values of fields are not defined - they are just strings... It depends only on implementation... If this driver i meant to work with XC NCIP toolkit based implementations, then I tried to fit to this library and it does not know the status "Accepted". That's why I changed this code. 

In fact we would need to make all these strings configurable... I's out of the scope of this PR for now I think...

Reply:
Makes sense to me! Just let me know when you would like to see this merged. Since these are primarily bug fixes, I could merge this now to get it included in the 7.0 release, but if you're planning to continue adding to this PR, I could also leave it open with the expectation that we'll make all the improvements at once in 7.1. I'm fine with either approach; just let me know what you prefer.

Reply:
I think it is  not necessary to have it in 7.0. I could continue to add some more tests, as these are only for LookupUser service. So, we can wait and merge later more work at once.

Reply:
Sounds good. I'll leave this open for now, and will await word from you when you are ready for further review and/or merging.